trunk/SUAVE/Components/Energy/Storages/Batteries/Constant_Mass/Lithium_Ion.py

```diff
@@ -1,33 +1,237 @@
 # ----------------------------------------------------------------------
 
 # suave imports
 import SUAVE
+from SUAVE.Core import Units, Data
+from SUAVE.Components.Energy.Storages.Batteries  import Battery  
 
 # package imports
-from SUAVE.Core import Units
-from SUAVE.Components.Energy.Storages.Batteries  import Battery
+import numpy as np 
+from scipy.integrate import  cumtrapz
 
 # ----------------------------------------------------------------------
 #  Lithium_Ion
 # ----------------------------------------------------------------------    
 ## @ingroup Components-Energy-Storages-Batteries-Constant_Mass
 class Lithium_Ion(Battery):
-    """
-    Specifies discharge/specific energy characteristics specific tobytes
-    lithium-ion batteries
-    """
+    """ Generic lithium ion battery that specifies  discharge/specific energy 
+    characteristics. 
+    
+    Assumptions
+    Default discharge curves correspond to lithium-iron-phosphate cells
+    
+    Inputs:
+    None
+    
+    Outputs:
+    None
+    
+    Properties Used:
+    N/A
+    """  
     def __defaults__(self):
-        self.specific_energy    = 200.    *Units.Wh/Units.kg
-        self.specific_power     = 1.      *Units.kW/Units.kg
-        self.ragone.const_1     = 88.818  *Units.kW/Units.kg
-        self.ragone.const_2     = -.01533 /(Units.Wh/Units.kg)
-        self.ragone.lower_bound = 60.     *Units.Wh/Units.kg
-        self.ragone.upper_bound = 225.    *Units.Wh/Units.kg
\ No newline at end of file
+            
+        self.tag               = 'Generic_Lithium_Ion_Battery_Cell'  
+        self.cell              = Data()   
+        self.module            = Data()        
+        self.pack_config       = Data()
+        self.module_config     = Data()
+        self.cooling_fluid     = Data()     
+        
+        self.cell.mass                                     = None
+        self.cell.charging_SOC_cutoff                      = 1. 
+        self.cell.charging_current                         = 3.0     # [Amps]
+        self.cell.charging_voltage                         = 3       # [Volts]
+                         
+        self.convective_heat_transfer_coefficient          = 35.     # [W/m^2K] 
+        self.heat_transfer_efficiency                      = 1.0       
+        
+        self.pack_config.series                            = 1
+        self.pack_config.parallel                          = 1  
+        self.pack_config.total                             = 1   
+        self.module_config.total                           = 1       
+        self.module_config.normal_count                    = 1       # number of cells normal to flow
+        self.module_config.parallel_count                  = 1       # number of cells parallel to flow      
+        self.module_config.normal_spacing                  = 0.02
+        self.module_config.parallel_spacing                = 0.02 
+
+        self.cooling_fluid.tag                             = 'air'
```

Comment:
pull from attributes so users can swap out fluids quickly

It's likely in the "future" (I know things) others will cool with fluids other than air....

Reply:
Updated.  Uses attributes class. Also cleaned up these scripts a bit